etl/src/error.rs

```diff
@@ -0,0 +1,898 @@
+        }
+    }
+}
+
+/// Converts [`rustls::Error`] to [`ETLError`] with [`ErrorKind::EncryptionError`].
+impl From<rustls::Error> for ETLError {
+    fn from(err: rustls::Error) -> ETLError {
+        ETLError {
+            repr: ErrorRepr::WithDescriptionAndDetail(
+                ErrorKind::EncryptionError,
+                "TLS configuration failed",
+                err.to_string(),
+            ),
+        }
+    }
+}
+
+/// Converts [`uuid::Error`] to [`ETLError`] with [`ErrorKind::InvalidData`].
+impl From<uuid::Error> for ETLError {
+    fn from(err: uuid::Error) -> ETLError {
+        ETLError {
+            repr: ErrorRepr::WithDescriptionAndDetail(
+                ErrorKind::InvalidData,
+                "UUID parsing failed",
+                err.to_string(),
+            ),
+        }
+    }
+}
+
+/// Converts [`chrono::ParseError`] to [`ETLError`] with [`ErrorKind::ConversionError`].
+impl From<chrono::ParseError> for ETLError {
+    fn from(err: chrono::ParseError) -> ETLError {
+        ETLError {
+            repr: ErrorRepr::WithDescriptionAndDetail(
+                ErrorKind::ConversionError,
+                "Chrono parse failed",
+                err.to_string(),
+            ),
+        }
+    }
+}
+
+/// Converts [`bigdecimal::ParseBigDecimalError`] to [`ETLError`] with [`ErrorKind::ConversionError`].
+impl From<bigdecimal::ParseBigDecimalError> for ETLError {
+    fn from(err: bigdecimal::ParseBigDecimalError) -> ETLError {
+        ETLError {
+            repr: ErrorRepr::WithDescriptionAndDetail(
+                ErrorKind::ConversionError,
+                "BigDecimal parsing failed",
+                err.to_string(),
+            ),
+        }
+    }
+}
+
+// SQLx error conversion
+
+/// Converts [`sqlx::Error`] to [`ETLError`] with appropriate error kind.
+///
+/// Maps database errors to [`ErrorKind::QueryFailed`], I/O errors to [`ErrorKind::IoError`],
+/// and connection pool errors to [`ErrorKind::ConnectionFailed`].
+impl From<sqlx::Error> for ETLError {
+    fn from(err: sqlx::Error) -> ETLError {
+        let kind = match &err {
+            sqlx::Error::Database(_) => ErrorKind::QueryFailed,
+            sqlx::Error::Io(_) => ErrorKind::IoError,
+            sqlx::Error::PoolClosed | sqlx::Error::PoolTimedOut => ErrorKind::ConnectionFailed,
+            _ => ErrorKind::QueryFailed,
+        };
+
+        ETLError {
+            repr: ErrorRepr::WithDescriptionAndDetail(
+                kind,
+                "Database operation failed",
+                err.to_string(),
+            ),
+        }
+    }
+}
+
+// BigQuery error conversions (feature-gated)
+
+/// Converts [`gcp_bigquery_client::error::BQError`] to [`ETLError`] with appropriate error kind.
+///
+/// Maps errors based on their specific type for better error classification and handling.
+#[cfg(feature = "bigquery")]
+impl From<gcp_bigquery_client::error::BQError> for ETLError {
```

Comment:
No action required in this PR right now but we should move the BigQuery destination outside the etl crate so that we code against the same etl interface that any third party destination writers are exposed to. This will also require some thought around how errors from those third party destinations work with etl.

Reply:
Errors from that crate could be handled with a classic enum type which is then converted into an `EtlError`.

Could you elaborate on why you would like to move them into a separate crate?

Reply:
Should have provided more context. Thought came to my mind from this issue: https://github.com/supabase/etl/issues/216. Basically anyone else who's writing their own destination currently runs into issues with reexports or other issues. If we have our destinations written in the exact same manner as any third party destination writers, by depending on an external crate, we're more likely to find and fix those issues.